app/scripts/components/modelcontrolmenu.js

```diff
@@ -81,13 +117,39 @@ var exports = (function () {
           this.deleteDialog.hide();
 
         };
-
-        // We also show an extra warning in the dialog, if user chooses to remove additional files.
         this.deleteDialog.showAlert(false);
 
         // Show the dialog:
         this.deleteDialog.show();
 
+      },
+
+      shareSelectedModels: function (domain) {
+        // Get a confirm dialog
+        this.shareDialog = getDialog(this, "confirm-dialog", "share-runs");
+
+        this.shareDialog.onConfirm = () => {
+          store.shareSelectedModels(domain);
+
+          this.shareDialog.hide();
+        };
+        this.shareDialog.showAlert(false);
+
+        // Show the dialog:
+        this.shareDialog.show();
+      },
+
+      downloadSelectedModels: function () {
+        store.downloadSelectedModels(this.selectedDownloads);
+      },
+
+      toggle: function(id, event) {
+        event.stopPropagation();
+        this.selectedDownloads[id] = !this.selectedDownloads[id];
+      },
+      preventCheck: function (event) {
+        event.preventDefault();
```

Comment:
default and propagation are handled by vue attributes in the click event handler `click.stop`